grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceAutoCompressTest.java

```diff
@@ -0,0 +1,101 @@
+import java.util.concurrent.LinkedTransferQueue;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.grpc.GrpcClients;
+import com.linecorp.armeria.common.grpc.GrpcSerializationFormats;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.grpc.testing.Messages.SimpleRequest;
+import com.linecorp.armeria.grpc.testing.Messages.SimpleResponse;
+import com.linecorp.armeria.grpc.testing.UnitTestServiceGrpc.UnitTestServiceBlockingStub;
+import com.linecorp.armeria.grpc.testing.UnitTestServiceGrpc.UnitTestServiceImplBase;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.logging.LoggingService;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import io.grpc.stub.StreamObserver;
+
+public class GrpcServiceAutoCompressTest {
+
+    @RegisterExtension
+    static final ServerExtension autoCompressionServer = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.workerGroup(1);
+            sb.requestTimeoutMillis(5000);
+            sb.serviceUnder("/",
+                            GrpcService.builder()
+                                       .autoCompression(true)
+                                       .addService(new UnitTestServiceImpl())
+                                       .build()
+                                       .decorate(LoggingService.newDecorator())
+                                       .decorate((delegate, ctx, req) -> {
+                                           ctx.log().whenComplete().thenAccept(requestLogQueue::add);
+                                           return delegate.serve(ctx, req);
+                                       }));
+        }
+    };
+
+    private static BlockingQueue<RequestLog> requestLogQueue = new LinkedTransferQueue<>();
+
+    @Test
+    void autoCompression() throws Exception {
+        requestLogQueue = new LinkedTransferQueue<>();
+        final UnitTestServiceBlockingStub client = GrpcClients.newClient(autoCompressionServer.httpUri(
+                GrpcSerializationFormats.PROTO).toString(), UnitTestServiceBlockingStub.class);
+        assertThat(client.staticUnaryCall(REQUEST_MESSAGE)).isEqualTo(RESPONSE_MESSAGE);
+        final RequestLog log = requestLogQueue.take();
+        assertThat(log.requestHeaders().get("grpc-accept-encoding")).isEqualTo("gzip");
+        assertThat(log.responseHeaders().get("grpc-encoding")).isEqualTo("gzip");
+    }
+
+    @Test
+    void autoCompressionWithMultipleAcceptEncoding() throws Exception {
+        requestLogQueue = new LinkedTransferQueue<>();
+        final UnitTestServiceBlockingStub client =
+                GrpcClients.builder(autoCompressionServer.httpUri(GrpcSerializationFormats.PROTO).toString())
+                           .addHeader("grpc-accept-encoding", "gzip,identity")
```

Comment:
Could we test this using `DecompressorRegistry`?
```java
var decompressorRegistry = 
  DecompressorRegistry
     .emptyInstance()
     .with(new Codec.Gzip(), true)
     .with(Codec.Identity.NONE, true);

builder.decompressorRegistry(decompressorRegistry);
```

Reply:
It seems that the settings of `DecompressorRegistry` don't affect `grpc-accept-encoding`.
So, I set `grpc-accept-encoding` directly.

Reply:
It sounds like a bug. I guess a `grpc-accept-encoding` should be made from the given `DecompressorRegistry`. Could you fix it together? 
https://github.com/line/armeria/blob/d32c7d8f1fa952ec00471812eb81ceaa4257a6a7/grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java#L109

Reply:
Sure, It should be fiexed.